From 46eb3706e5fd5733dfd1681d99c31e86e760ddd6 Mon Sep 17 00:00:00 2001
From: "Kothapeta, BikshapathiX" <bikshapathix.kothapeta@intel.com>
Date: Fri, 9 Oct 2020 16:37:54 +0530
Subject: [PATCH] [SEpolicy] provide rw permissions for sysfs

allow vendor_init permission for sysfs_net:file r_file_perms;
and update genfs_contexts as per adb shell ls -al /sys/class/net/
eth0 -> ../../devices/pci0000:00/0000:00:04.0/net
wlan0 -> ../../devices/pci0000:00/0000:00:07.0/net

Signed-off-by: Kothapeta, BikshapathiX <bikshapathix.kothapeta@intel.com>

diff --git a/ethernet/common/genfs_contexts b/ethernet/common/genfs_contexts
index f7b75f6..93f9386 100644
--- a/ethernet/common/genfs_contexts
+++ b/ethernet/common/genfs_contexts
@@ -1,3 +1,5 @@
 # /sys/class/net
 genfscon sysfs /devices/pci0000:00/0000:00:13.2/0000:03:00.0/net u:object_r:sysfs_net:s0
 genfscon sysfs /devices/pci0000:00/0000:00:13.1/0000:02:00.0/net u:object_r:sysfs_net:s0
+genfscon sysfs /devices/pci0000:00/0000:00:04.0/net u:object_r:sysfs_net:s0
+genfscon sysfs /devices/pci0000:00/0000:00:07.0/net u:object_r:sysfs_net:s0
diff --git a/ethernet/common/vendor_init.te b/ethernet/common/vendor_init.te
new file mode 100644
index 0000000..7c6fba6
--- /dev/null
+++ b/ethernet/common/vendor_init.te
@@ -0,0 +1 @@
+allow vendor_init sysfs_net:file r_file_perms;
--
2.29.2

